build/patches/Add-a-proxy-configuration-page-fix-377-679.patch

```diff
@@ -0,0 +1,236 @@
+From: root <root@ubuntu.cabponte>
+Date: Sun, 23 Aug 2020 20:00:18 +0000
+Subject: fix 377 and 679
+
+---
+ chrome/browser/resources/proxy_config.html |  5 ++-
+ chrome/browser/resources/proxy_config.js   | 24 +++++++----
+ chrome/browser/ui/webui/proxy_config_ui.cc | 46 +++++++++++++++-------
+ net/proxy_resolution/proxy_config.cc       | 16 +++++---
+ 4 files changed, 63 insertions(+), 28 deletions(-)
+ mode change 100644 => 100755 chrome/browser/resources/proxy_config.html
+ mode change 100644 => 100755 chrome/browser/resources/proxy_config.js
+ mode change 100644 => 100755 chrome/browser/ui/webui/proxy_config_ui.cc
+ mode change 100644 => 100755 net/proxy_resolution/proxy_config.cc
+
+diff --git a/chrome/browser/resources/proxy_config.html b/chrome/browser/resources/proxy_config.html
+old mode 100644
+new mode 100755
+index 898716564728630258b4d4c442d21da9d326f1a1..1db228c3bbfe52c35adea95a36fe8190a07d6cbb
+--- a/chrome/browser/resources/proxy_config.html
++++ b/chrome/browser/resources/proxy_config.html
+@@ -44,14 +44,15 @@
+         Reset will update the displayed configuration to match the one currently in use.
+       </div>
+       <div class="section-container">
+-          <input type="radio" id="empty" name="mode" value="empty" checked><label for="empty">None</label><br/>
++          <input type="radio" id="empty" name="mode" value="empty"><label for="empty">System Default</label><br/>
++          <input type="radio" id="direct" name="mode" value="direct"><label for="direct">Direct</label><br/>
+           <input type="radio" id="auto-detect" name="mode" value="auto-detect"><label for="auto-detect">Auto-detect (WPAD DHCP/DNS)</label><br/>
+           <input type="radio" id="use-pac-url" name="mode" value="use-pac-url"><label for="use-pac-url">Use PAC URL: <input id='pac-url' value="" size="40" /></label>
+           <p><input type="checkbox" id="pac-mandatory" name="pac-mandatory"><label for="pac-mandatory">Do not allow fallback to direct connection in case PAC script fails</label></p>
+           <input type="radio" id="use-single-list" name="mode" value="use-single-list"><label for="use-single-list">Use a single proxy list for all schemes (<a target="_blank" href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Proxy_servers_and_tunneling/Proxy_Auto-Configuration_(PAC)_file#Description">PAC format</a>):
+               <textarea id="single-proxies"></textarea>
+             </label><br/>
+-          <input type="radio" id="use-list-per-scheme" name="mode" value="use-list-per-scheme"><label for="use-list-per-scheme">Use a proxy list per scheme:</label><br/>
++          <input type="radio" id="use-list-per-scheme" name="mode" value="use-list-per-scheme"><label for="use-list-per-scheme">Use a proxy list per scheme (use ; as separator):</label><br/>
```

Comment:
Also newlines work? I would omit this, otherwise it would be needed on every box.

Reply:
I think it's needed. Is one of the things I would ask myself right away, what do I delimit it from? what kinda of programmers couldn't write it? now I have to go to the site to find out if it's a bug or what! :)

Reply:
You still need to visit the documentation to see how the PAC format works.


If you want to add it, shouldn't the label for the other boxes also mention it?

Reply:
"without directions, always straight."

> If you want to add it, shouldn't the label for the other boxes also mention it?
as you like, no important

Reply:
I don't understand the quote; I removed it from the version that will be used.

Reply:
> I removed it from the version that will be used.

ok